### ISSUE TITLE
How dose TrainingA and TrainingB work in Training phase?/Losses in custom training dataset are not stable.

### ISSUE THREAD

**Issue by KoyuruKU** [nixingyang/FlipReID#13]
Hello.
Thank you for publishing your project.
I have actually used it and was amazed at the accuracy.

I have two questions.
The first question is how is TrainingA different from TrainingB in the Training phase and how does it work?

My second question is that I added a new dataset to the Market1501 dataset and trained it, but the loss is not stable.
Do you know why the loss is unstable as shown in the attached graph?
The attached graphs are the loss of TrainingA and the loss of TrainingB.
I used ResNest as the backbone model and trained TrainingA with 20 epoc and TrainingB with 200 epoc.

Thank you.
![loss_trainingA](https://user-images.githubusercontent.com/124120439/215974182-6eb2129d-56b3-4a27-b703-f1a497a84b6b.png)
![loss_trainingB](https://user-images.githubusercontent.com/124120439/215974187-4fc57559-7be4-4af1-953d-8822b428d2be.png)

**Comment by nixingyang** [comment]
Hi,
Thanks for your interest in our work.

- Q1: how is TrainingA different from TrainingB in the Training phase and how does it work?
- A1: You can find a detailed comparison between Figure 2 (a) and (b) in [Section 2.2](https://arxiv.org/pdf/2105.05639.pdf).
- Q2: I added a new dataset to the Market1501 dataset and trained it, but the loss is not stable. Do you know why the loss is unstable as shown in the attached graph? The attached graphs are the loss of TrainingA and the loss of TrainingB. I used ResNest as the backbone model and trained TrainingA with 20 epoc and TrainingB with 200 epoc.
- A2: What you have reported is a bit unexpected. Are you sure that the only change is the dataset? On the one hand, you can run experiments on Market-1501 and verify that both settings work properly. On the other hand, you may reduce the learning rate in `TrainingA`.

All the best.
Xingyang

**Comment by KoyuruKU** [comment]
Thank you for your quick reply.

Regarding A1, I recognized that in TrainingA, the baseline processing is done as shown in Figure 2(a) in the paper, and in TrainingB, the processing is done as shown in Figure 2(b) with the addition of the Flipping process. 

As you pointed out about A2, I will investigate this while adjusting the learning rate.

Thank you very much.

**Comment by KoyuruKU** [comment]
Sorry for asking so many questions.

I do not understand the role of the classification model in the paper.

In **2.2. FlipReID** of the paper, **Options for Training and Inference.** it says that the classification models are discarded during inference, but what is the reason for creating classification models during training?
Also, in **Sub-Models.** in **2.1 Baseline**, the layer **BatchNormalizationDense** in Figure 1 is described as a classification model, but is it correct to assume that this layer is not used during inference?

I apologize if I have skipped some parts of the paper.
Thank you

**Comment by nixingyang** [comment]
Hi,

- Q1: What is the reason for creating classification models during training?
- A1: It is a very common practice to adopt both the batch hard triplet loss and categorical cross-entropy loss in optimizing a person re-identification model. Without the categorical cross-entropy loss, training is likely to be unstable and the resulting model will have inferior performance. You may conduct some experiments to verify this claim.
- Q2: Is it correct to assume that this layer is not used during inference?
- A2: Yes. In the inference procedure, we only need to extract the feature vector. These classification models would be discarded.

All the best.
Xingyang

**Comment by KoyuruKU** [comment]
Thank you for answering my question.

I found out about the difference between the classification model and the inference model.
I will try to find some more information and learn more on my own.
If I have any more questions I will ask you.

Thank you very much.
